src/ucs/config/parser.c

```diff
@@ -1169,14 +1169,27 @@ static ucs_status_t ucs_config_apply_env_vars(void *opts, ucs_config_field_t *fi
     return UCS_OK;
 }
 
+static const char * ucs_config_parser_get_base_prefix(const char *env_prefix)
+{
+    int sub_prefix_len;
+    
+    /* env_prefix always has "_" at the end and we want to find the last but one
+     * "_" in the env_prefix */
+    sub_prefix_len = strlen(env_prefix) - 2;
+
+    while ((sub_prefix_len >= 0) && (env_prefix[sub_prefix_len] != '_')) {
+        sub_prefix_len -= 1;
+    }
+    return env_prefix + sub_prefix_len;
```

Comment:
1. space line before
2. what if sub_prefix_len < 0? let's return NULL in this case

Reply:
It's actually copy-paste bug, sub_prefix_len cannot be negative.
There should be sub_prefix_len += 1 after while loop